apps/backend/src/web/api/v2/util.ts

```diff
@@ -152,11 +152,34 @@ const createBuild = async (params: {
   return build;
 };
 
+/**
+ * Special token used to reference the reference branch of the project.
+ * Interpreted by Argos SDKs especially for GitHub Actions with "deployment_status" event.
+ */
+const REFERENCE_BRANCH = "__argos/reference-branch";
+
+/**
+ * Resolves the reference branch if a special token is used.
+ */
+async function resolveReferenceBranch(
+  referenceBranch: string,
+  project: Project,
+) {
+  if (referenceBranch === REFERENCE_BRANCH) {
+    return project.$getReferenceBranch();
+  }
+
+  return null;
```

Comment:
The previous value was `req.body.referenceBranch ?? null`.

The new value is `referenceBranch === REFERENCE_BRANCH ? project.$getReferenceBranch() : null`

It should be : `referenceBranch === REFERENCE_BRANCH ? project.$getReferenceBranch() : referenceBranch`